extensions/amp-auto-ads/0.1/firstimpression.io-network-config.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Services} from '../../../src/services';
+import {buildUrl} from '../../../ads/google/a4a/shared/url-builder';
+import {dict} from '../../../src/utils/object';
+
+/**
+ * @implements {./ad-network-config.AdNetworkConfigDef}
+ */
+export class FirstImpressionIoConfig {
+  /**
+   * @param {!Element} autoAmpAdsElement
+   */
+  constructor(autoAmpAdsElement) {
+    this.autoAmpAdsElement_ = autoAmpAdsElement;
+  }
+
+  /**
+   * @param {!Window} unused
+   * @override
+   */
+  isEnabled(unused) {
+    return true;
+  }
+
+  /**
+   * True if responsive is enabled for auto-ads
+   */
+  isResponsiveEnabled() {
+    return false;
+  }
+
+  /** @override */
+  getConfigUrl() {
+    let previewId = 0;
+
+    const previewFlowRegex = /amp\/fi\/(\d+)\//;
+    const previewFlowParam = window.location.pathname.match(previewFlowRegex);
```

Comment:
We want to avoid any direct access to window, as it may not do what you expect in contexts like the amp viewer. `this.win` is available, or the `Url` service.

Reply:
Same question as above - will it be OK to use `this.autoAmpAdsElement_.ampdoc_.win.location.pathname`? Thanks.